units/host_monitoring_idle_termination.go

```diff
@@ -166,6 +166,35 @@ func (j *idleHostJob) Run(ctx context.Context) {
 }
 
 func (j *idleHostJob) checkAndTerminateHost(ctx context.Context, h *host.Host) error {
+
+	communicationTime, idleTime, err := j.getTimesFromHost(ctx, h)
+
+	if err != nil {
+		return errors.Wrapf(err, "error getting communication and idle time from '%s'", h.Id)
+	}
+
+	idleThreshold := idleTimeCutoff
+	if h.RunningTaskGroup != "" {
+		idleThreshold = idleTaskGroupHostCutoff
+	}
+
+	// if we haven't heard from the host or it's been idle for longer than the cutoff, we should terminate
+	terminateReason := ""
+
+	if communicationTime >= idleThreshold {
+		terminateReason = fmt.Sprintf("host is idle or unreachable, communication time %d over threshold minutes %d", communicationTime, idleThreshold)
+	} else if idleTime >= idleThreshold {
+		terminateReason = fmt.Sprintf("host is idle or unreachable, idleTime %d over threshold minutes %d", idleTime, idleThreshold)
+	}
+	if terminateReason != "" {
+		j.Terminated++
+		j.TerminatedHosts = append(j.TerminatedHosts, h.Id)
+		return j.env.RemoteQueue().Put(ctx, NewHostTerminationJob(j.env, h, false, terminateReason))
+	}
+	return nil
+}
+
+func (j *idleHostJob) getTimesFromHost(ctx context.Context, h *host.Host) (time.Duration, time.Duration, error) {
```

Comment:
What you did by making the new job depend on this job was very clever! I actually thought this could just be a helper function on the host or something though (or just not a method). Do we use the job here at all besides logging the ID? It could make things simpler to just log inside the original function so you don't need to rely on a job at all (although you've already done the refactor so whatever you prefer now really!)